tv-shows-app/src/components/shared/SidebarNavigation/SideBarNavigation.tsx

```diff
@@ -0,0 +1,111 @@
+"use client";
+import { Flex, Text } from "@chakra-ui/react";
+import NextLink from "next/link";
+import { useEffect, useState } from "react";
+import { PiTelevisionSimple } from "react-icons/pi";
+import { FiLogOut } from "react-icons/fi";
+
+export default function SidebarNavigation() {
+  const link = window.location.href;
+  const currentCat = link.split("/").reverse()[0];
+
+  const [category, setCategory] = useState(currentCat);
+
+  useEffect(() => {
+    const handleUrlChange = () => {
+      const link = window.location.href;
+      const newCategory = link.split("/").reverse()[0];
+      setCategory(newCategory);
+    };
+
+    window.addEventListener("popstate", handleUrlChange);
+
+    return () => {
+      window.removeEventListener("popstate", handleUrlChange);
+    };
+  }, []);
+
+  return (
+    <Flex
+      bg="purple.800"
+      minHeight="100vh"
+      height="auto"
+      width="20vw"
+      color="white"
+      flexDirection="column"
+      padding={10}
+      gap={3}
+      fontSize="large"
+      position="fixed"
+      boxShadow="10px 0px 10px rgba(0, 0, 0, 0.3)"
+    >
+      <Flex marginLeft={-5} marginTop={-5} marginBottom={5}>
+        <PiTelevisionSimple size={30} />
+        <Text
+          marginLeft={2}
+          letterSpacing="wide"
+          fontSize="xl"
+          fontWeight="bold"
+          color="purple.100"
+          textShadow="2px 2px 4px rgba(0, 0, 0, 0.5)"
+        >
+          TV SHOWS APP
+        </Text>
+      </Flex>
+
+      <Text
```

Comment:
An improvement on this would be to have a `NavigationLink` component which would get the current category and handle the conditional styling internally so you don't repeat it three (or more when you get more routes) times. It would be great to see how you would implement this.